rosidl_typesupport_connext_c/src/wstring_conversion.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <rosidl_typesupport_connext_c/wstring_conversion.hpp>
+
+#include "ndds/ndds_c.h"
+#include "rosidl_generator_c/u16string_functions.h"
+
+namespace rosidl_typesupport_connext_c
+{
+
+DDS_Wchar * create_wstring_from_u16string(const rosidl_generator_c__U16String & u16str)
+{
+  DDS_Wchar * wstr = DDS_Wstring_alloc(u16str.size);
```

Comment:
Assuming this can fail, guard against a failure to allocate.

Reply:
Done: 6fb05a09dfd93df184cd42630dfb90333c62b9d0.